app/src/main/java/com/github/palFinderTeam/palfinder/ui/login/LoginActivity.kt

```diff
@@ -1,37 +1,53 @@
 package com.github.palFinderTeam.palfinder.ui.login
 
 import android.content.Intent
+import android.content.IntentSender
 import android.os.Bundle
 //import android.text.Editable
 //import android.text.TextWatcher
 import android.util.Log
+import android.widget.FrameLayout
 //import android.widget.EditText
 //import android.widget.Toast
 //import androidx.annotation.StringRes
 import androidx.appcompat.app.AppCompatActivity
 import com.github.palFinderTeam.palfinder.MainActivity
 import com.github.palFinderTeam.palfinder.R
+import com.github.palFinderTeam.palfinder.utils.PrettyDate
+import com.google.android.gms.auth.api.identity.BeginSignInRequest
+import com.google.android.gms.auth.api.identity.Identity
+import com.google.android.gms.auth.api.identity.SignInClient
 import com.google.android.gms.auth.api.signin.GoogleSignIn
 import com.google.android.gms.auth.api.signin.GoogleSignInOptions
 import com.google.android.gms.common.SignInButton
 import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.CommonStatusCodes
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.auth.FirebaseUser
 import com.google.firebase.auth.GoogleAuthProvider
 import com.google.firebase.auth.ktx.auth
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.SetOptions
+import com.google.firebase.firestore.ktx.firestore
 import com.google.firebase.ktx.Firebase
+import java.security.Timestamp
+import java.util.*
 
 
 class LoginActivity : AppCompatActivity() {
 
     private lateinit var auth: FirebaseAuth
     //private lateinit var signInButton: FrameLayout
+    private lateinit var oneTapClient: SignInClient
+    private lateinit var signInRequest: BeginSignInRequest
+
 
     private companion object{
         private const val TAG = "LoginActivity"
         private const val RC_GOOGLE_SIGN_IN = 4926
-        //Google service id
-        private const val GID = "341371843047-6i3a92lfmcb6555vsj9sb02tnhmkh4c8.apps.googleusercontent.com"
+        private val REQ_ONE_TAP = 2  // Can be any integer unique to the Activity
+        private var showOneTapUI = true
+        val db = Firebase.firestore
```

Comment:
Indeed a memory leak, you can directly call Firebase.firestore in your code, it always return the same instance. In any case we will probably provide a clean interface later this week, to add, modify, and delete a profile, which we probably will use here too.